internal/surfer/gcloud/builder.go

```diff
@@ -0,0 +1,403 @@
+		// For example, in a `CreateInstance` method, this would be the `instance_id` field.
+		if utils.IsPrimaryResource(field, method) {
+			// If it is the primary resource, we generate a special positional argument for it.
+			param := newPrimaryResourceParam(field, method, model, overrides, service)
+			args.Params = append(args.Params, param)
+			continue
+		}
+
+		// For all other fields, we generate a standard flag argument. If the field
+		// is a nested message, its fields will be "flattened" into top-level flags.
+		// For example, a field `instance.description` becomes the `--description` flag.
+		if err := addFlattenedParams(field, field.JSONName, &args, overrides, model, service); err != nil {
+			return Arguments{}, err
+		}
+	}
+	return args, nil
+}
+
+// addFlattenedParams recursively processes a field and its sub-fields to generate
+// a flat list of command-line flags. This is necessary for nested messages in
+// the request proto.
+func addFlattenedParams(field *api.Field, prefix string, args *Arguments, overrides *Config, model *api.API, service *api.Service) error {
+	// We skip fields that are marked as `OUTPUT_ONLY` in the proto, as these are
+	// not meant to be provided by the user. We also skip the "name" field, as it's
+	// handled by the primary resource argument.
+	if slices.Contains(field.Behavior, api.FIELD_BEHAVIOR_OUTPUT_ONLY) || field.Name == "name" {
+		return nil
+	}
+
+	// If the field is a nested message (and not a map, which is handled differently),
+	// we need to recurse into its fields. This is the "flattening" process.
+	// For example, in the Parallelstore API, the `CreateInstanceRequest` message
+	// has a field named `instance` which is of type `Instance`. The `Instance`
+	// message itself has fields like `description` and `capacity_gib`.
+	// This block will recurse into the `Instance` message's fields.
+	if field.MessageType != nil && !field.Map {
+		for _, f := range field.MessageType.Fields {
+			// The prefix is updated to create a dot-separated path for the `api_field`.
+			// Continuing the example: when processing the `capacity_gib` field inside the
+			// `Instance` message, the prefix will become "instance.capacityGib". This
+			// results in a `--capacity-gib` flag that maps to the correct nested field.
+			if err := addFlattenedParams(f, fmt.Sprintf("%s.%s", prefix, f.JSONName), args, overrides, model, service); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	// If the field is a scalar, map, or enum, we generate a parameter for it.
+	param, err := newParam(field, prefix, overrides, model, service)
+	if err != nil {
+		return err
+	}
+	args.Params = append(args.Params, param)
+	return nil
+}
+
+// newParam creates a single command-line argument (a `Param` struct) from a proto field.
+func newParam(field *api.Field, apiField string, overrides *Config, model *api.API, service *api.Service) (Param, error) {
```

Comment:
Nit: To avoid this function becoming a "mutation soup," we should eventually move toward a registry-dispatch pattern (or some logic that makes input, analysis, and output easy to debug and add to). It will become hard to track how the Param state changes across nested if-else blocks, which makes it fragile (e.g., changing a field for update commands might accidentally break create).

Checking with rust and golang team for advice would be good. For now, this is fine for a POC and is more important for phase 3